main.py

```diff
@@ -59,186 +60,193 @@ def main():
+                                    'symbol': announcement_coin,
+                                    'price': price,
+                                    'volume': qty,
+                                    'time': datetime.timestamp(datetime.now()),
+                                    'tp': tp,
+                                    'sl': sl,
+                                    'id': 'test-order',
+                                    'text': 'test-order',
+                                    'create_time': datetime.timestamp(datetime.now()),
+                                    'update_time': datetime.timestamp(datetime.now()),
+                                    'currency_pair': f'{announcement_coin}_{pairing}',
+                                    'status': 'filled',
+                                    'type': 'limit',
+                                    'account': 'spot',
+                                    'side': 'buy',
+                                    'iceberg': '0'
+                                }
+                                logger.info('PLACING TEST ORDER')
+                                logger.debug(order[announcement_coin])
+                            # place a live order if False
+                            else:
+                                logger.info(f'starting buy place_order with : {announcement_coin=} | {pairing=} | {qty=} | side = buy | {price=}')
+                                order[announcement_coin] = place_order(announcement_coin, pairing, qty,'buy', price)
+                                order[announcement_coin] = order[announcement_coin].__dict__
+                                order[announcement_coin].pop("local_vars_configuration")
+                                order[announcement_coin]['tp'] = tp
+                                order[announcement_coin]['sl'] = sl
+                                logger.info('Finished buy place_order')
+
+                        except Exception as e:
+                            logger.error(e)
 
+                        else:
+                            logger.info(f'Order created with {qty} on {announcement_coin}')
+                            store_order('order.json', order)
                     else:
-                        logger.info(f'Order created with {qty} on {announcement_coin}')
-                        store_order('order.json', order)
-                else:
-                    if announcement_coin:
                         logger.warning(f'{announcement_coin=} is not supported on gate io')
                         if os.path.isfile('new_listing.json'):
                             os.remove("new_listing.json")
                             logger.debug('Removed new_listing.json due to coin not being '
-                                      'listed on gate io')
+                                    'listed on gate io')
+                else:
+                    logger.error('supported_currencies is not initialized')
```

Comment:
get_all_currecies() getting called here could have caused our main thread to get stuck in an endless loop if supported_currencies is False for some reason.